packages/datagateway-dataview/src/views/card/cardView.component.tsx

```diff
@@ -0,0 +1,814 @@
+        return null;
+      case undefined:
+        return 'asc';
+    }
+  };
+
+  React.useEffect(() => {
+    // console.log('Page number (page): ', page);
+    // console.log('Current pageNum (query): ', query.page);
+    // console.log('Page change: ', pageChange);
+    // console.log('Page results: ', maxResults);
+    // console.log('Query results: ', query.results);
+
+    // Set the page number if it was found in the parameters.
+    if (!pageChange) {
+      if (query.page) {
+        // console.log('Set to query page: ', query.page);
+        setPage(query.page);
+      } else {
+        // Workaround for issue where page remains same on pagination on investigation/dataset.
+        // If this is not a page change and there is no page query parameter,
+        // then default to the 1st page (we treat this as the initial page load).
+        setPage(1);
+      }
+    } else {
+      // Manually scroll to top of the page as the pagination click isn't doing so.
+      window.scrollTo(0, 0);
+      setPageChange(false);
+    }
+
+    // Ensure the max results change according to the query parameter.
+    if (query.results && resOptions.includes(query.results)) {
+      // dataCount > resOptions[0]
+      // maxResults !== query.results
+      setMaxResults(query.results);
+    } else {
+      // Reset the max results back to the default value
+      // when switching between pages (this is the same issue as
+      // the pagination, page, holding the same value between
+      // investigation/dataset card views).
+      setMaxResults(resOptions[0]);
+    }
+  }, [page, pageChange, query, maxResults, dataCount, resOptions]);
+
+  // TODO: Work-around for the pagination, start/stop index
+  //       working incorrectly due to the totalDataCount being updated later on
+  //       (to the new value for the new view).
+  React.useEffect(() => {
+    // Handle count and reloading of data based on pagination options.
+    if (totalDataCount > 0) {
+      setDataCount(totalDataCount);
+      // Calculate the maximum pages needed for pagination.
+      const p = ~~((totalDataCount + maxResults - 1) / maxResults);
+
+      // Check if the page query is correct, if not then set the page back to 1.
+      if (query.page && query.page > p) {
+        changePage(1);
+      }
+      setNumPages(p);
+      // console.log('num pages: ', numPages);
+      setLoadedData(false);
+    }
+  }, [changePage, maxResults, numPages, query.page, totalDataCount]);
+
+  // TODO: Creates duplicate count request.
+  // TODO: This should be not how filter/sort changes work; make it simpler (may require a big change).
+  React.useEffect(() => setFilterChange(true), [filters]);
+  React.useEffect(() => setSortChange(true), [sort]);
+
+  // TODO: Duplicate count and data requests for filter and page changes.
+  React.useEffect(() => {
+    // TODO: Move this separately so that sort and filters are handled separately
+    //       (tied to only one load count/data).
```

Comment:
@patrick-austin this is one of the main items to fix with the card view. It is really an issue only because of how the different `useEffects` and updated props are interacting with each other.